### PR TITLE
Mean IoU Implementation

### LINKED ISSUE #4394: Implementation of IoU
**Is your feature request related to a problem? Please describe.**
Please include the IoU as additional counting based metric

**Describe the solution you'd like**
IoU is simply defined as TP/(2*TP+FP+FN) - However one should indicate point out in the documentation that IoU and F1 evaluate similar aspects and should therefore not be used in conjunction

### PR DESCRIPTION
**Pull request by yashika-git**
Fixes #4394

### Description
Implementation of IoU Metric.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [x] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: docs/source/metrics.rst [hunk 1]
```diff
@@ -39,6 +39,13 @@ Metrics
 .. autoclass:: DiceMetric
     :members:
 
+`Mean IoU`
+-----------
```
Comment (Nic-Ma) [inline docs/source/metrics.rst:43]: Please ensure the length of `---` matches the string.

Thanks.

File: docs/source/metrics.rst [hunk 2]
```diff
@@ -39,6 +39,13 @@ Metrics
...
+.. autofunction:: compute_meaniou
```
Comment (Nic-Ma) [inline docs/source/metrics.rst:44]: @wyli I think `compute_iou` may be better? No need `mean`?

Thanks.

**Review by Nic-Ma — APPROVED** [review]
Looks good to me, put minor comments inline.
Please fix the flake8 format issue in CI tests.
@KumoLiu Could you please help double confirm the computation logic of IoU here?

Thanks in advance.

File: docs/source/metrics.rst [hunk 3]
```diff
@@ -39,6 +39,13 @@ Metrics
...
+
+.. autoclass:: IoUMetric
```
Comment (Nic-Ma) [inline docs/source/metrics.rst:46]: To be consistent with the function, the metric class name may be `MeanIoU`?

Thanks.

File: docs/source/metrics.rst (hunk 1 above)
Comment (yashika-git) [reply docs/source/metrics.rst]: Done!

File: docs/source/metrics.rst (hunk 3 above)
Comment (yashika-git) [reply docs/source/metrics.rst:46]: Okay, I'll do this

**Comment by yashika-git** [comment]
@Nic-Ma, could you please help with the failing tests?
Thank you

**Comment by KumoLiu** [comment]
Hi @Nic-Ma,
 I have checked the computation logic of IoU here. It looks good to me!
Thanks!

**Review by Nic-Ma — APPROVED** [review]
Thanks for the quick update and @KumoLiu 's confirmation.
It looks good to me.
@wyli Do you have any other comments? If not, I merge it soon.

Thanks.

**Pull request merged**